Adds set up app links cookbook (#8374)
flutter/website: src/cookbook/navigation/set-up-app-links.md

        ),
      ],
    );
    ```

## 2. Modify AndroidManifest.xml

1. Open the Flutter project with Android Studio or VSCode. 
2. Navigate to `android/app/src/main/AndroidManifest.xml` file
3. Add the following metadata tag and intent filter inside the `<activity>` tag with `.MainActivity`

    Replace `example.com` with your own web domain.
    ```
    <meta-data android:name="flutter_deeplinking_enabled" android:value="true" />
    <intent-filter android:autoVerify="true">
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <data android:scheme="http" android:host="example.com" />
        <data android:scheme="https" />
    </intent-filter>
    ```
   {{site.alert.note}}
     The metadata tag flutter_deeplinking_enabled opts into Flutter's default deeplink handler. If
     you are using the third-party plugins, such as [uni_links][], setting this metadata tag will
     break these plugins. Omit this metadata tag if you prefer to use third-party plugins.
   {{site.alert.end}}

## 3. Hosting assetlinks.json file

> You need to host an `assetlinks.json` file in the web domain.

Reviewer: I think this should say:

```suggestion
Host an `assetlinks.json` file in using a web server with a domain that you own.
```

This file tells the mobile browser which Android application to open instead of the browser.
To create the file, get the package name of the Flutter app you created in the previous step and
the sha256 fingerprint of the signing key you will be using to build the APK.

### Package name

Locate the package name in `AndroidManifest.xml`, the `package` property under `<manifest>` tag.
Package names are usually in the format of `com.example.*`.

### sha256 fingerprint

> Process may differ depending on the way apk is signed.

Reviewer: Can we provide a link?

#### Using google play app signing

> You can find the sha256 fingerprint directly from play console under `Setup > App Integrity`

Reviewer: A screenshot might be helpful here.

#### Using local keystore

> If you stores key locally, you can generate sha256 using the following command

Reviewer: ```suggestion
If you are storing the key locally, you can generate sha256 using the following command.
```

```
keytool -list -v -keystore <path-to-keystore>
```

### assetlinks.json

> The hosted file should have the following content:

Reviewer: Will the fingerprints be different?

```suggestion
The hosted file should look similar to this:
```

  }
}]
```

1. Set the `package_name` value to your Flutter application package name.

2. Set sha256_cert_fingerprints to the value you got from the previous step.

3. Host the file at a URL that resembles the following:
`<webdomain>/.well-known/assetlinks.json`

4. Verify that your browser can access this file.

## Testing

You can use a real device or the Emulator to test a app link,
but first make sure you have executed `flutter run` at least once on
the devices. This ensures that the Flutter application is installed.

<img src="/assets/images/docs/cookbook/set-up-app-links-emulator-installed.png" alt="Emulator screenshot" width="50%" />

To test **only** the app setup, use the adb command:
```
adb shell 'am start -a android.intent.action.VIEW \
    -c android.intent.category.BROWSABLE \
    -d "http://<web-domain>/details"' \
    <package name>
```
{{site.alert.note}}
   This does not test whether the web files are hosted correctly, the command launches the app even
   if web files are not presented.
{{site.alert.end}}

To test **both** web and app setup, you need to click on a link directly through web browser or other

> apps. One way is to use the google doc by typing the url, and click.

Reviewer: ```suggestion
apps. One way is to create a Google Doc, add the link, and tap on it.
```
